ios/NativeSignerTests/Core/Database/DatabaseMediatorTests.swift

```diff
@@ -0,0 +1,308 @@
+
+        // When
+        _ = subject.recreateDatabaseFile()
+
+        // Then
+        XCTAssertEqual(fileManager.urlForInAppropriateForCreateCallsCount, 1)
+        XCTAssertEqual(fileManager.urlForInAppropriateForCreateReceivedDirectory, [.documentDirectory])
+        XCTAssertEqual(fileManager.urlForInAppropriateForCreateReceivedDomain, [.userDomainMask])
+        XCTAssertEqual(fileManager.urlForInAppropriateForCreateReceivedShouldCreate, [false])
+    }
+
+    func test_recreateDatabaseFile_whenFileExists_removesFileAtExpectedDestination() {
+        // Given
+        let expectedPathUrl = fileManager.urlForInAppropriateForCreateReturnValue.appendingPathComponent("Database")
+        fileManager.fileExistsAtPathReturnValue = true
+
+        // When
+        _ = subject.recreateDatabaseFile()
+
+        // Then
+        XCTAssertEqual(fileManager.removeItemAtCallsCount, 1)
+        XCTAssertEqual(fileManager.removeItemAtReceivedURL, [expectedPathUrl])
+    }
+
+    func test_recreateDatabaseFile_whenFileDoesNotExist_doesNotAttemptToRemoveFile() {
+        // Given
+        fileManager.fileExistsAtPathReturnValue = false
+
+        // When
+        _ = subject.recreateDatabaseFile()
+
+        // Then
+        XCTAssertEqual(fileManager.removeItemAtCallsCount, 0)
+    }
+
+    func test_recreateDatabaseFile_whenDestinationAccessThrowsError_returnsFalse() {
+        // Given
+        fileManager.urlForInAppropriateForCreateThrowableError = ErrorMock.unknown
+        fileManager.removeItemAtThrowableError = nil
+
+        // When
+        let result = subject.recreateDatabaseFile()
+
+        // Then
+        XCTAssertFalse(result)
+    }
+
+    func test_recreateDatabaseFile_whenFileExists_whenRemoveItemThrowsError_returnsFalse() {
+        // Given
+        fileManager.fileExistsAtPathReturnValue = true
+        fileManager.urlForInAppropriateForCreateThrowableError = nil
+        fileManager.removeItemAtThrowableError = ErrorMock.unknown
+
+        // When
+        let result = subject.recreateDatabaseFile()
+
+        // Then
+        XCTAssertFalse(result)
+    }
+
+    func test_recreateDatabaseFile_whenCopyItemThrowsError_returnsFalse() {
+        // Given
+        fileManager.urlForInAppropriateForCreateThrowableError = nil
+        fileManager.copyItemAtToThrowableError = ErrorMock.unknown
+
+        // When
+        let result = subject.recreateDatabaseFile()
+
+        // Then
+        XCTAssertFalse(result)
+    }
+
+    func test_recreateDatabaseFile_whenNoThrownErrors_returnsTrue() {
+        // Given
+        fileManager.urlForInAppropriateForCreateThrowableError = nil
+        fileManager.removeItemAtThrowableError = nil
+        fileManager.copyItemAtToThrowableError = nil
+
+        // When
+        let result = subject.recreateDatabaseFile()
+
+        // Then
+        XCTAssertTrue(result)
+    }
+}
+
+// MARK: - Mocks
+
+final class FileManagingProtocolMock: FileManagingProtocol {
```

Comment:
For now I'm creating these mocks within my won personal project that has [Sourcery](https://github.com/krzysztofzablocki/Sourcery) setup for mock generation, I'll remove those after Sourcery is integrated within Signer and those mocks can be autogenerated locally during compilation time, so no need to check them out into the repo. 
They will follow exact same syntax that you can see for below mocks 🙏🏻